android/MobiledgeXSDKDemo/app/src/main/java/com/mobiledgex/sdkdemo/MatchingEngineHelper.java

```diff
@@ -80,8 +88,204 @@ public MatchingEngineHelper(Context context, View view) {
         mContext = context;
         mView = view;
         mMatchingEngine = new MatchingEngine(mContext);
+        mMatchingEngine.setSSLEnabled(false); //TODO: Remove. Only here to connect to DIND DME
+
+        // Register ourselves. The Subscribe annotation will be called on ClientEdgeEvents.
+        mMatchingEngine.getEdgeEventBus().register(this);
+        Log.i(TAG, "Subscribed to ServerEdgeEvents");
+    }
+
+    /**
+     * Subscribe to ServerEdgeEvents! (Guava Interface)
+     * To optionally post messages to the DME, use MatchingEngine's DMEConnection.
+     */
+    @Subscribe
+    public void onMessageEvent(AppClient.ServerEdgeEvent event) {
+        Log.i(TAG, "onMessageEvent: "+event);
+        Map<String, String> tagsMap = event.getTagsMap();
+
+        switch (event.getEventType()) {
+            case EVENT_INIT_CONNECTION:
+                Log.i(TAG, "Received Init response: " + event);
+                mMatchingEngineResultsInterface.showMessage("Received: " + event.getEventType());
+                break;
+            case EVENT_APPINST_HEALTH:
+                Log.i(TAG, "Received: AppInst Health: " + event);
+                handleAppInstHealth(event);
+                break;
+            case EVENT_CLOUDLET_STATE:
+                Log.i(TAG, "Received: Cloutlet State event: " + event);
+                handleCloudletState(event);
+                break;
+            case EVENT_CLOUDLET_MAINTENANCE:
+                Log.i(TAG, "Received: Cloutlet Maintenance event." + event);
+                handleCloudletMaintenance(event);
+                break;
+            case EVENT_LATENCY_PROCESSED:
+                Log.i(TAG, "Received: Latency has been processed on server: " + event);
+                handleEventLatencyResults(event);
+                break;
+            case EVENT_LATENCY_REQUEST:
+                Log.i(TAG, "Received: Latency has been requested to be tested (client perspective): " + event);
+                handleLatencyRequest(event);
+                break;
+            case EVENT_CLOUDLET_UPDATE:
+                Log.i(TAG, "Received: Server pushed a new FindCloudletReply to switch to: " + event);
+                handleFindCloudletServerPush(event);
+                break;
+            case EVENT_UNKNOWN:
+                Log.i(TAG, "Received UnknownEvent.");
+                break;
+            default:
+                Log.i(TAG, "Event Received: " + event.getEventType());
+        }
+        // TODO: Need event switch of some kind to handle.

```

Comment:
Remove. There's a terminate event now.

Reply:
@lgarner, what are you suggesting to remove? The highlighted code above, or the `if (tagsMap.containsKey("shutdown"))` block?